packages/shared/components/organisms/AccountActivity.svelte

```diff
@@ -21,6 +27,55 @@
         })
     }
 
+    async function _syncAccount() {
+        $isSyncing = true
+        try {
+            await get(selectedAccount).sync()
+            $isSyncing = false
+        } catch (err) {
+            $isSyncing = false
+
+            const shouldHideErrorNotification =
+                err && err.type === 'ClientError' && err.error === 'error.node.chrysalisNodeInactive'
+            if (!shouldHideErrorNotification) {
+                if ($isLedgerProfile) {
+                    displayNotificationForLedgerProfile('error', true, true, false, false, err)
+                } else {
+                    showAppNotification({
+                        type: 'error',
+                        message: localize(err.error),
+                    })
+                }
+            }
+        }
+    }
+
+    async function handleSyncAccountClick() {
+        if (!$isSyncing) {
+            if ($isSoftwareProfile) {
+                try {
+                    const strongholdStatusResponse = await get(profileManager).isStrongholdPasswordAvailable()
+
+                    if (strongholdStatusResponse) {
+                        openPopup({
+                            type: 'password',
+                            props: { onSuccess: () => _syncAccount() },
+                        })
```

Comment:
I think we have an abstraction for this called checkStronghold response or something